### PR TITLE
Do not set `use_ios_simulator` unconditionally

### PR DESCRIPTION
**Pull request by dnfield**
@zanderso 

This causes gn to emit warnings on non-mac platforms

### CODE REVIEW

**Review by zanderso — CHANGES_REQUESTED** [review]
`use_ios_simulator` defaults to true, and messes up Android builds on mac.

**Comment by dnfield** [comment]
Ok, I think I figured out what's going on here -updated the `else` to only target mac.

File: tools/gn [hunk 1]
```diff
@@ -303,7 +303,7 @@ def to_gn_args(args):
 
   if gn_args['target_os'] == 'ios':
     gn_args['use_ios_simulator'] = args.simulator
-  else:
+  elif gn_args['target_os'] == 'mac':
```
Comment (zanderso) [inline tools/gn:306]: It's a bit more complicated since you can target Android while on a mac.

Comment (dnfield) [reply tools/gn:306]: But simulator is iOS specific right?

**Comment by dnfield** [comment]
Filed https://github.com/flutter/flutter/issues/105476 for the infra flakes.

File: tools/gn [hunk 2]
```diff
@@ -303,7 +303,7 @@ def to_gn_args(args):
 
   if gn_args['target_os'] == 'ios':
     gn_args['use_ios_simulator'] = args.simulator
-  else:
+  elif gn_args['host_os'] == 'mac':
```
Comment (zanderso) [inline tools/gn:306]:
```suggestion
  elif get_host_os() == 'mac':
```

**Review by zanderso — COMMENTED** [review]
gn_args['host_os'] isn't always filled in apparently.

**Review by zanderso — APPROVED** [review]

**Comment by zanderso** [comment]
Thanks!

**Pull request merged**